src/unistd.rs

```diff
@@ -986,6 +986,20 @@ fn pipe2_setflags(fd1: RawFd, fd2: RawFd, flags: OFlag) -> Result<()> {
     }
 }
 
+/// Truncate a file to a specified length
+///
+/// See also
+/// [truncate(2)](http://pubs.opengroup.org/onlinepubs/9699919799/functions/truncate.html)
+pub fn truncate<P: ?Sized + NixPath>(path: &P, len: off_t) -> Result<()> {
+    let res = try!(path.with_nix_path(|cstr| {
+        unsafe {
+            libc::truncate(cstr.as_ptr(), len)
+        }
+    }));
+
+    Errno::result(res).map(drop)
```

Comment:
I prefer `map(|| ())` for stylistic reasons.  `drop`, in this context, is basically being called for its side effect of returning nothing.

Reply:
I'm not exactly sure what you'd like to see here, but given that this is a copy/paste from the many, many other functions that follow this style, I think keeping this for consistency is good.